sklego/model_selection.py

```diff
@@ -0,0 +1,43 @@
+from sklearn.model_selection._split import _BaseKFold
+import numpy as np
+import pandas as pd
+
+
+class KlusterFoldValidation(_BaseKFold):
+    """
+    K-Means cross validator
+
+    :param n_splits: Number of splits (clusters) to fold on
+    :param cluster_method: Clustering method with fit_predict attribute
+    """
+
+    def __init__(self, random_state, cluster_method=None):
```

Comment:
hows about we pass an object into the initializer. 

1. this object can be an untrained clustering method
2. this object can be a trained clustering method 

either way ... this might solve lots of issues when you want to 

1. allow for a pipeline
2. allow for all sorts of settings